BDArmory/Radar/VesselRadarData.cs

```diff
@@ -1853,11 +1853,10 @@ public void UnlockCurrentTarget()
         public void UnlockSelectedTarget(Vessel vessel)
         {
             if (!locked) return;
-
             var vesselIndex = displayedTargets.FindIndex(t => t.vessel == vessel);
             if (vesselIndex != -1 && vesselIndex < lockedTargetIndexes.Count && lockedTargetIndexes[vesselIndex] >= 0 && lockedTargetIndexes[vesselIndex] < displayedTargets.Count) // FIXME These shouldn't be necessary and are likely from a bug!
```

Comment:
Using `vesselIndex` instead of `lockedTargetIndexes[vesselIndex]` means that only the `if (vesselIndex != -1)` check should be required again.